registry/servicediscovery/openapi.yml

```diff
@@ -0,0 +1,59 @@
+definitions:
+    RegistryPortNumber:
+        format: int64
+        type: integer
+    RegistryServiceDiscoveryResponse:
```

Comment:
The existing library does have a _little_ more structure than just `map[string]any`, though indeed it does require the caller to know which of the two currently-supported "shapes" a particular service keyword uses:

* [`Host.ServiceURL`](https://pkg.go.dev/github.com/hashicorp/terraform-svchost@v0.1.1/disco#Host.ServiceURL) asserts that the value ought to be a string containing a URL. The library automatically parses the URL and, if it's relative, resolves it to an absolute URL using the discovery document's own URL as the base. This one is the main one used for most services, including `modules.v1` and `providers.v1`.
* [`Host.ServiceOAuthClient`](https://pkg.go.dev/github.com/hashicorp/terraform-svchost@v0.1.1/disco#Host.ServiceOAuthClient) deals with the specialized structure that's currently used only by `login.v1` to represent a full OAuth client configuration. As with `ServiceURL` this automatically parses and resolves URLs relative to the discovery document's URL.

A typical pattern for using it would therefore be something like:

```go
services := disco.New()
host, err := disco.Discover(providerAddr.Host)
// (error handling)
registryBaseURL, err := host.ServiceURL("providers.v1")
// (error handling)
client := NewProviderRegistryClient(registryBaseURL)
```

...so there's no `any` exposed in the public API intended for use in client configuration, despite the fact that it does of course use such a representation a little internally. (You _can_ find it exposed in the `Disco.ForceHostServices` but that's an OpenTofu CLI configuration concern -- the undocumented `host` block for overriding host services during development -- and so not really in scope for what this library is about.)

The spec you've put in this package _does_ explicitly encode the fact that `providers.v1` exists and its raw value is a string, but since it just exposes the raw URLs strings without parsing/resolving them I assume the client would be responsible for the URL resolution here, and they don't seem to actually have enough information to perform that step because the final service discovery document URL (after following any redirects) isn't exposed anywhere.

I guess a way to get "the best of both worlds" here would be to retain this package's awareness of the specific services `providers.v1`, `modules.v1`, and `login.v1` but to bring in the logic that parses and resolves the URLs to absolute `*url.URL`, instead of just exposing the raw URL strings.